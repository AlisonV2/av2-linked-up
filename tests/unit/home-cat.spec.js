import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import HomeCategories from '@/components/home/Categories';
import AppTitle from '@/components/app/Title';

const actions = {
  getArtistsByStyle: jest.fn(),
};

const store = createStore({
  actions,
});

const $router = {
  push: jest.fn(),
};

/**
 * @module HomeCategoriesTest
 */
describe('HomeCategories.vue', () => {
  it('setCategory event should be called', async () => {
    const component = shallowMount(HomeCategories, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
          'app-title': AppTitle,
        },
        mocks: {
          $router,
        },
      },
    });
    component.find('[data-test="black-work"]').trigger('click');
    await component.vm.$nextTick();
    expect(component.emitted().setCategory).toHaveBeenCalled;
  });
  it('Router push should be called', async () => {
    const component = shallowMount(HomeCategories, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
          'app-title': AppTitle,
        },
        mocks: {
          $router,
        },
      },
    });
    await component.find('[data-test="black-work"]').trigger('click');
    expect($router.push).toHaveBeenCalledWith({ name: 'CategoriesResults' });
  });
  it('getArtistsByStyle should be called', async () => {
    const component = shallowMount(HomeCategories, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
          'app-title': AppTitle,
        },
        mocks: {
          $router,
        },
      },
    });
    component.find('[data-test="black-work"]').trigger('click');
    await component.vm.$nextTick();
    expect(actions.getArtistsByStyle).toHaveBeenCalled();
  });
});
