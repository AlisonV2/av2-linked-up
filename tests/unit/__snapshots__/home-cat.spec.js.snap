// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCategories.vue HomeCategories Snapshot 1`] = `
<div
  class="row"
>
  <div
    class="col-12 text-center mb-4 app-title"
  >
    <h2
      class="title"
    >
      <img
        class="title-img"
        src="/img/title-bg.png"
      />
      <span>
        
        Search by style
        
      </span>
    </h2>
  </div>
  <div
    class="col-12 col-md-6 col-lg-3"
  >
    <div
      class="card"
      data-test="black-work"
    >
      <img
        alt="blackwork"
        class="card-img-top"
        src="/img/blackwork.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Black Work
        </h5>
      </div>
    </div>
  </div>
  <div
    class="col-12 col-md-6 col-lg-3"
  >
    <div
      class="card"
    >
      <img
        alt="oldschool"
        class="card-img-top"
        src="/img/oldschool.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Old School
        </h5>
      </div>
    </div>
  </div>
  <div
    class="col-12 col-md-6 col-lg-3"
  >
    <div
      class="card"
    >
      <img
        alt="newschool"
        class="card-img-top"
        src="/img/newschool.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          New School
        </h5>
      </div>
    </div>
  </div>
  <div
    class="col-12 col-md-6 col-lg-3"
  >
    <div
      class="card"
    >
      <img
        alt="surrealism"
        class="card-img-top"
        src="/img/surrealism.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Surrealism
        </h5>
      </div>
    </div>
  </div>
  <div
    class="col-12 text-end"
  >
    <a>
      <p
        class="link-text"
      >
        See all styles
      </p>
    </a>
  </div>
</div>
`;
